optimum/exporters/openvino/convert.py

```diff
@@ -1188,3 +1201,43 @@ def get_flux_models_for_export(pipeline, exporter, int_dtype, float_dtype):
         models_for_export["text_encoder_2"] = (text_encoder_2, export_config)
 
     return models_for_export
+
+
+def _get_encoder_decoder_stateful_models_for_export(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    task: str,
+    _variant: str,
+    library_name: str,
+    int_dtype: str = "int64",
+    float_dtype: str = "fp32",
+    preprocessors: Optional[List[Any]] = None,
+):
+    export_config_constructor = TasksManager.get_exporter_config_constructor(
+        model=model, exporter="openvino", task=task, library_name=library_name
+    )
+    export_config = export_config_constructor(
+        model.config,
+        int_dtype=int_dtype,
+        float_dtype=float_dtype,
+        preprocessors=preprocessors,
+        legacy=False,
+    )
+
+    export_config.variant = _variant
+    all_variants = "\n".join([f"    - {name}: {description}" for name, description in export_config.VARIANTS.items()])
+    logger.info(f"Using the export variant {export_config.variant}. Available variants are:\n{all_variants}")
+
+    models_for_export = _get_submodels_for_export_encoder_decoder(model, use_past=True)
```

Comment:
looks like we could also set `use_past=False` and use  `models_for_export[DECODER_NAME]` directly no ? 
https://github.com/huggingface/optimum/blob/d21256c2964945fc3fe4623f7befb21082b69a25/optimum/exporters/utils.py#L203

Reply:
not sure that I understand your suggestion, we still need past as outputs and inputs for making transformation (their filling just will be hided inside model and performed automatically, but for applying our transformations we need them anyway), that is why decoder_with_past is more suitable for such case than just decoder.

I think I can figure out how to achieve the same effect with model without past (changing decoder config during export)

Reply:
When calling `_get_submodels_for_export_encoder_decoder(model, use_past=True)` it looks like the models `models_for_export[DECODER_NAME]` and `models_for_export[DECODER_WITH_PAST_NAME]` will be equivalent 

https://github.com/huggingface/optimum/blob/d21256c2964945fc3fe4623f7befb21082b69a25/optimum/exporters/utils.py#L213-L215

so  I was suggesting to set `use_past=False` and use  `models_for_export[DECODER_NAME]` directly, would that make sense ?

Reply:
yes, thanks,I applied your suggestion and it looks better now